core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/ProducerProperties.java

```diff
@@ -81,6 +81,11 @@ public class ProducerProperties {
 
 	private PollerProperties poller;
 
+	/**
+	 * A flag which enables/disables kafka partition count updates during runtime.
+	 */
+	private boolean kafkaPartitionCountUpdateEnabled = false;
```

Comment:
Actually, since this is a binder-specific feature, I would suggest making this part of the extended binding properties in `KafkaProducerProperties.` We don't want to leak this field to the core `ProducerProperties.` Also, how about `dynamicPartitionUpdatesEnabled` for the property name? Or something similar? Once the changes are in the `KafkaProducerProperties`, we don't need the prefix `kafka`.

Reply:
I'm not sure if it is in general really a Kafka related feature, but there will be only my Kafka implementation for now. KafkaProducerProperties are mostly related to real Kafka properties from DefaultKafkaProducerFactory, which is not the case here. I rename it to  'dynamicPartitionUpdatesEnabled' and it is currently only implemented for KafkaBinders. I'm not sure about other technologies and if a partitionCount could also be changed at runtime. This should a topic for a architecture discussion.

Reply:
Ok, that is a fair point. Since it is adding to the core producer properties, let me discuss this with our internal engineering team. I will get back to you asap.

Reply:
Will do that.